src/py/pyodide/http.py

```diff
@@ -148,8 +151,64 @@ async def bytes(self) -> bytes:
         self._raise_if_failed()
         return (await self.buffer()).to_bytes()
 
+    async def _into_file(self, f: IOBase):
+        """Write the data into an empty file with no copy.
 
-async def pyfetch(url, **kwargs) -> FetchResponse:
+        Warning: should only be used when f is an empty file, otherwise it may
+        overwrite the data of f.
+        """
+        buf = await self.buffer()
+        buf._into_file(f)
+
+    async def create_file(self, path: str):
+        """Uses the data to back a new file without copying it.
+
+        This method avoids copying the data when creating a new file. If you
+        want to write the data into an existing file, use
+
+        .. code-block:: python
+
+            buf = await resp.buffer()
+            buf.to_file(file)
```

Comment:
The method name is not consistent in this example with the one above I think?

Also I think `to_file` rather than `create_file` would be better, to be constent with the buffer method name

Reply:
Well this creates a new file, whereas the buffer method appends to an existing file. I think they need to be named differently since they do different things.

Reply:
Usually, that kind of difference would be passed via the `mode="w"` or `mode='a'` parameter and I find it somewhat confusing to have different names for those. Also, a method that fails when a file exists (instead of overwriting it) is pretty much the opposite of what Python usually does by default?

For instance https://docs.rs/serde_any/0.1.2/serde_any/fn.to_file.html there is no other method for append/create as far as I can tell.
I understand there are implementation constraints but from a usability perspective, I find these different names fairly confusing in the public API.

Could we benchmark how much time we save as opposed to `buf.to_file` with an extra copy? As if the HTTP request is 100ms and the extra copy is 100ns maybe making API more complex for it it's not worth it.

Reply:
> Also, a method that fails when a file exists (instead of overwriting it) is pretty much the opposite of what Python usually does by default?

I am happy to make it overwrite by default. I guess I am not that familiar with the typical Python behaviors.

>  maybe making API more complex for it it's not worth it.

Probably yeah. Seems to take about 1ms to copy a 300kb buffer, about 3ms to copy a 1mb buffer.